rust-sdk/whirlpool/src/config.rs

```diff
@@ -9,6 +9,41 @@ pub const DEFAULT_WHIRLPOOLS_CONFIG_ADDRESS: Pubkey = Pubkey::new_from_array([
     209, 37, 94, 53, 123, 51, 119, 221, 238, 28, 205,
 ]);
 
+/// Predefined networks for WhirlpoolsConfig.
+pub enum Network {
+    SolanaMainnet,
+    SolanaDevnet,
+    EclipseMainnet,
+    EclipseTestnet,
+}
+
+impl Network {
+    pub fn config_address(&self) -> Pubkey {
+        match self {
+            Network::SolanaMainnet => DEFAULT_WHIRLPOOLS_CONFIG_ADDRESS,
+            Network::SolanaDevnet => Pubkey::new_from_array([
```

Comment:
In this case probably not that big of a deal in this case, but since rust is a compiled language there is a performance improvement we can make by defining the addresses as `const`. A `const` is evaluated at compile-time instead of at runtime meaning that they are faster at runtime.
```rust
const SOLANA_MAINNET_WHIRLPOOLS_CONFIG_ADDRESS = Pubkey::new_from_array([...]);
const SOLANA_DEVNET_WHIRLPOOLS_CONFIG_ADDRESS = Pubkey::new_from_array([...]);
const ECLIPSE_MAINNET_WHIRLPOOLS_CONFIG_ADDRESS = Pubkey::new_from_array([...]);
const ECLIPSE_TESTNET_WHIRLPOOLS_CONFIG_ADDRESS = Pubkey::new_from_array([...]);

impl Network {
  pub fn config_address(&self) -> Pubkey {
    match self {
      Network::SolanaMainnet => SOLANA_MAINNET_WHIRLPOOLS_CONFIG_ADDRESS,
      Network::SolanaDevnet => SOLANA_DEVNET_WHIRLPOOLS_CONFIG_ADDRESS,
      Network::EclipseMainnet => ECLIPSE_MAINNET_WHIRLPOOLS_CONFIG_ADDRESS,
      Network::EclipseTestnet => ECLIPSE_TESTNET_WHIRLPOOLS_CONFIG_ADDRESS,
    }
  }
}
```